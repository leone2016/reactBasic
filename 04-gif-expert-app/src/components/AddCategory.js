import React, { useState } from 'react'
import PropTypes from 'prop-types';

 const AddCategory = (props) => {
    const { setcategories } = props;
    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) => {
        setInputValue(e.target.value);
        console.log('handleInputChange Called');
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        if( inputValue.trim().length > 2 ){
            setcategories(cat=> [ inputValue, ...cat]);
            setInputValue('');
        }
       
    }
    return (
        <form onSubmit={handleSubmit}>
            <p> {inputValue} </p>
            <input
                type="text"
                value={inputValue}
                onChange={handleInputChange}
            />
        </form>
    )
}
AddCategory.prototype = {
    setcategories : PropTypes.func.isRequired
}


export default AddCategory;
