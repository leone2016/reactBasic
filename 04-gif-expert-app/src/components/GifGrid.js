import React from 'react';
import {GifGridItem } from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';
/* import { getGifs } from '../helpers/getGifs'; */


export const GifGrid = ({ category }) => {

    const {data: images,  loading } = useFetchGifs(category);

   /*  const [images, setImages] = useState([]) */

  

    return (
        <>
            <h3>{category}</h3>
            <div className="card-grid">

            {loading && <h4>C A R G A N D O . . .</h4>}
                <ol>
                   {
                        images.map(img => (
                            <GifGridItem
                                key={img.id}
                                {...img}
                            />
                        ))
                    }
                </ol>
           
            </div>
        </>
    )
}
