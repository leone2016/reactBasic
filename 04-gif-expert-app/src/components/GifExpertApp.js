import React, { useState } from 'react'
import AddCategory from './AddCategory';
import { GifGrid } from './GifGrid';

const GifExpertApp = props => {
    const initialState = ['One Punch'];
    const [categories, setcategories] = useState(initialState);

    const handleAdd = (category) => {
        /* setcategories( ['HunterHunter', ...categories]); */
    }
    return (
        <div>
            <h2>GifExpertApp</h2>
            <AddCategory setcategories={setcategories} />
            <hr />
            <button onClick={handleAdd}>Agregar</button>
            <ol>
                {
                    categories.map((cat) => (
                      <GifGrid
                      key={cat}
                      category={cat}
                      />
                    )

                    )
                }
            </ol>
        </div>
    )
}

export default GifExpertApp
