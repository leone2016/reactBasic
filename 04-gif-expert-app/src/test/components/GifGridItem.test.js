import React from 'react';
import { shallow } from "enzyme";
import '@testing-library/jest-dom';
import { GifGridItem } from "../../components/GifGridItem";

describe('UNIT TEST GifGridItem.test.js', () => {

    const title = 'un t√¨tulo';
    const url = 'https://imagentest.jpg';
    const wrapper = shallow(<GifGridItem title={title} url={url} />);

    test('should be show <GifGridItem/> correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should be have a paragraph with the title ', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    });

    test('should be have an image with url and alt equals to props', () => {
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    });

    test('should be have animate__fadeIn', () => {
        const cardDiv = wrapper.find('div');
        console.log(cardDiv.props());
        const className = cardDiv.prop('className');
        expect(className.includes('animate__fadeIn')).toBe(true);
    });
});
