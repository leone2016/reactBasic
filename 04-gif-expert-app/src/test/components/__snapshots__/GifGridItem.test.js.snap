// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UNIT TEST GifGridItem.test.js should be show <GifGridItem/> correctly 1`] = `
<div
  className="card animate__fadeIn"
>
  <img
    alt="un tìtulo"
    src="https://imagentest.jpg"
  />
  <p>
    un tìtulo
  </p>
</div>
`;
